File: schemaorg-rs/src/category_code.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type CategoryCodeAdditionalTypeFieldEnum = String;
///<https://schema.org/CategoryCode>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct CategoryCode {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/inCodeSet>
    #[serde(rename = "inCodeSet")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub in_code_set: Vec<CategoryCodeInCodeSetFieldEnum>,
    ///<https://schema.org/codeValue>
    #[serde(rename = "codeValue")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub code_value: Vec<String>,
    ///<https://schema.org/inDefinedTermSet>
    #[serde(rename = "inDefinedTermSet")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub in_defined_term_set: Vec<CategoryCodeInDefinedTermSetFieldEnum>,
    ///<https://schema.org/termCode>
    #[serde(rename = "termCode")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub term_code: Vec<String>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub main_entity_of_page: Vec<CategoryCodeMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub identifier: Vec<CategoryCodeIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub description: Vec<CategoryCodeDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub subject_of: Vec<CategoryCodeSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub additional_type: Vec<CategoryCodeAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    #[serde(default)]
    pub image: Vec<CategoryCodeImageFieldEnum>,
}

File: schemaorg-ts/src/category_code.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type CategoryCodeAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/CategoryCode>
 **/
export interface CategoryCode {
  "@context": string;
  "@type": "CategoryCode";

  /**
   * <https://schema.org/inCodeSet>
   **/
  inCodeSet: all.OneOrMany<all.CategoryCodeInCodeSetFieldEnum>;

  /**
   * <https://schema.org/codeValue>
   **/
  codeValue: all.OneOrMany<string>;

  /**
   * <https://schema.org/inDefinedTermSet>
   **/
  inDefinedTermSet: all.OneOrMany<all.CategoryCodeInDefinedTermSetFieldEnum>;

  /**
   * <https://schema.org/termCode>
   **/
  termCode: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.CategoryCodeMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.CategoryCodeIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.CategoryCodeDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.CategoryCodeSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.CategoryCodeAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.CategoryCodeImageFieldEnum>;
}
